exporter/otlp-grpc/lib/opentelemetry/exporter/otlp/grpc/exporter.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/common'
+require 'opentelemetry/exporter/otlp/common'
+require 'opentelemetry/sdk'
+require 'grpc'
+
+require 'google/rpc/status_pb'
+require 'opentelemetry/proto/collector/trace/v1/trace_service_services_pb'
+
+module OpenTelemetry
+  module Exporter
+    module OTLP
+      module GRPC
+        # An OpenTelemetry trace exporter that sends spans over GRPC.
+        class Exporter
+          SUCCESS = OpenTelemetry::SDK::Trace::Export::SUCCESS
+          FAILURE = OpenTelemetry::SDK::Trace::Export::FAILURE
+          private_constant(:SUCCESS, :FAILURE)
+
+          def initialize(endpoint: config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_EXPORTER_OTLP_ENDPOINT', default: 'http://localhost:4317/v1/traces'),
+                         timeout: config_opt('OTEL_EXPORTER_OTLP_TRACES_TIMEOUT', 'OTEL_EXPORTER_OTLP_TIMEOUT', default: 10),
+                         metrics_reporter: nil)
+            raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}" if invalid_url?(endpoint)
+
+            uri = URI(endpoint)
+
+            @client = Opentelemetry::Proto::Collector::Trace::V1::TraceService::Stub.new(
+              "#{uri.host}:#{uri.port}",
+              :this_channel_is_insecure
```

Comment:
Quite literally a copy paste from the quick start example for GRPC.

https://grpc.io/docs/languages/ruby/quickstart/#update-the-client

This field is required, but I'm unfamiliar with how configuration should interact with it so I'm leaving it as is (because it works with a local collector running).